Editor/Scripts/CodeGenerators/RGActionCodeGenerator.cs

```diff
@@ -100,6 +106,103 @@ private static void SearchForBotActionMethods()
+                    string className = classDeclaration.Identifier.ValueText;
+                    List<RGStateInfo> stateList = new List<RGStateInfo>();
+
+                    var membersWithRGState = classDeclaration.Members
+                        .Where(m => m.AttributeLists.Any(a => a.Attributes.Any(attr => attr.Name.ToString() == "RGState")));
+
+                    foreach (var member in membersWithRGState)
+                    {
+                        string fieldType = member is MethodDeclarationSyntax ? "method" : "variable";
+                        string fieldName = member is MethodDeclarationSyntax
+                            ? ((MethodDeclarationSyntax) member).Identifier.ValueText
+                            : ((FieldDeclarationSyntax) member).Declaration.Variables.First().Identifier.ValueText;
+
+                        string stateName = fieldName;
+                        var attribute = member.AttributeLists.SelectMany(attrList => attrList.Attributes)
+                            .FirstOrDefault(attr => attr.Name.ToString() == "RGState");
+
+                        if (attribute != null)
+                        {
+                            var attributeArgument = attribute.ArgumentList?.Arguments.FirstOrDefault();
+                            if (attributeArgument != null &&
+                                attributeArgument.Expression is LiteralExpressionSyntax literal)
+                            {
+                                stateName = literal.Token.ValueText;
+                            }
+                        }
+
+                        string type = member is MethodDeclarationSyntax
+                            ? RemoveGlobalPrefix(semanticModel.GetTypeInfo(((MethodDeclarationSyntax) member).ReturnType)
+                                .Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat))
+                            : RemoveGlobalPrefix(semanticModel.GetTypeInfo(((FieldDeclarationSyntax) member).Declaration.Type)
+                                .Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat));
+
+                        stateList.Add(new RGStateInfo
+                        {
+                            FieldType = fieldType,
+                            FieldName = fieldName,
+                            StateName = stateName,
+                            Type = type
+                        });
+                    }
+
+                    if (stateList.Any())
+                    {
+                        rgStateInfoList.Add(new RGStatesInfo
+                        {
+                            Object = className,
+                            State = stateList
+                        });
+                    }
+                }
+            }
+
+            string jsonResult = JsonConvert.SerializeObject(new { RGStateInfo = rgStateInfoList }, Formatting.Indented);
+            
+            // TODO (REG-1267): send json result to server for typedef generation
```

Comment:
Is this still a TODO? It seems like rn we are fine with the typdef generation happening here? Or by typedef do you mean code generation for bots, not generating the JSON?

Reply:
Actually for now, because we're doing user upload of the data, it would be something like:
```
// TODO (REG-1267) Export JSON for user upload
```
The ticket: https://linear.app/regression-games/issue/REG-1267/unity-agent-builder-crossover-export-actions-and-state

Reply:
Ah ok, thanks!